generated/nitclk/unit_tests/test_nitclk.py

```diff
@@ -25,10 +33,202 @@ def teardown_method(self, method):
+        min_time = datetime.timedelta(seconds=0.042)
+        self.patched_library.niTClk_Synchronize.side_effect = self.side_effects_helper.niTClk_Synchronize
+        nitclk.synchronize(multiple_sessions, min_time)
+        self.patched_library.niTClk_Synchronize.assert_called_once_with(_matchers.ViUInt32Matcher(len(multiple_sessions)), _matchers.ViSessionBufferMatcher(multiple_sessions), _matchers.ViReal64Matcher(min_time.total_seconds()))
+        return
+
+    def test_synchronize_to_sync_pulse_sender(self):
+        min_time = 0.042
+        self.patched_library.niTClk_SynchronizeToSyncPulseSender.side_effect = self.side_effects_helper.niTClk_SynchronizeToSyncPulseSender
+        nitclk.synchronize_to_sync_pulse_sender(multiple_sessions, min_time)
+        self.patched_library.niTClk_SynchronizeToSyncPulseSender.assert_called_once_with(_matchers.ViUInt32Matcher(len(multiple_sessions)), _matchers.ViSessionBufferMatcher(multiple_sessions), _matchers.ViReal64Matcher(min_time))
         return
 
+    def test_wait_until_done(self):
+        min_time = 4.2
+        self.patched_library.niTClk_WaitUntilDone.side_effect = self.side_effects_helper.niTClk_WaitUntilDone
+        nitclk.wait_until_done(multiple_sessions, min_time)
+        self.patched_library.niTClk_WaitUntilDone.assert_called_once_with(_matchers.ViUInt32Matcher(len(multiple_sessions)), _matchers.ViSessionBufferMatcher(multiple_sessions), _matchers.ViReal64Matcher(min_time))
+        return
+
+    # API error handling
+
+    def test_api_error(self):
+        error_string = 'Error'
+        self.patched_library.niTClk_Synchronize.side_effect = self.side_effects_helper.niTClk_Synchronize
+        self.side_effects_helper['Synchronize']['return'] = -1
+        self.patched_library.niTClk_GetExtendedErrorInfo.side_effect = self.side_effects_helper.niTClk_GetExtendedErrorInfo
+        self.side_effects_helper['GetExtendedErrorInfo']['errorString'] = error_string
+        try:
+            nitclk.synchronize(multiple_sessions, 0.42)
+        except nitclk.Error as e:
+            assert e.code == -1
+            assert e.description == error_string
+
+    def test_api_get_error_description_fails(self):
+        self.patched_library.niTClk_Synchronize.side_effect = self.side_effects_helper.niTClk_Synchronize
+        self.side_effects_helper['Synchronize']['return'] = -1
+        self.patched_library.niTClk_GetExtendedErrorInfo.side_effect = self.side_effects_helper.niTClk_GetExtendedErrorInfo
+        self.side_effects_helper['GetExtendedErrorInfo']['return'] = -2
+        try:
+            nitclk.synchronize(multiple_sessions, 0.42)
+        except nitclk.Error as e:
+            assert e.code == -1  # we want the original error code from getting the attribute.
+            assert e.description == "Failed to retrieve error description."
+
+    # SessionReference error handling
+
+    def test_session_reference_error(self):
+        session = nitclk.SessionReference(SESSION_NUM_FOR_TEST)
```

Comment:
Why? What do we gain by this? It add complexity.

Reply:
I didn't know having more than 1 test class adds complexity. Could you elaborate? It seems to me better organized to break the nitclk function tests from the SessionReference class tests.

Reply:
Because mocking nitclk dll has to happen in multiple test classes and I was not able to get mocking one DLL in multiple instances to work in the past. It would require figuring out what is going on and working around it. If it is even possible.